app/controllers/sessions_controller.rb

```diff
@@ -30,20 +30,24 @@ def gitlab
   end
 
   def failure
+    Samson::Hooks.fire(:audit_action, nil, 'login', :failed, auth_hash)
```

Comment:
I made the audit_action method support more than 3 args for such cases in the future, but this could be simplified to failed login, successful login, etc to eliminate the 3rd arg.